eox_tenant/backends/database.py

```diff
@@ -180,3 +180,156 @@ def set_key_to_cache(self, key, value):
+            candidates = cached_list
+        else:
+            candidates = self.backend_manager.objects.all()  # pylint: disable=no-member
+            self.set_key_to_cache(cache_key, candidates)
+
+        for microsite in candidates:
+            yield microsite
+
+    def get_config_by_domain(self, domain):
+        """
+        Get the correct set of site configurations.
+        """
+        from eox_tenant.models import TenantConfig
+        configurations, external_key = TenantConfig.get_configs_for_domain(domain)
+
+        if not (configurations and external_key):
+            configurations, external_key = self._get_microsite_config_by_domain(domain)
+
+        return configurations, external_key
+
+    def _get_microsite_config_by_domain(self, domain):
+        """
+        Return the configuration and key available for a given domain.
+        """
+        from eox_tenant.models import Microsite
+        microsite = Microsite.get_microsite_for_domain(domain)
+
+        if microsite:
+            return microsite.values, microsite.key
+
+        return {}, None
+
+    def get_all_orgs(self):
+        """
+        Return a set of orgs that are considered within all microsites.
+        This can be used, for example, to do filtering
+        """
+        org_filter_set = super(TenantConfigCompatibleMicrositeBackend, self).get_all_orgs()
+
+        # Get the orgs in the tenant config db model
+        for microsite in self.iterate_sites_with_tenant():
+            current = microsite.lms_configs
+            org_filter = current.get('course_org_filter')
+            if org_filter and isinstance(org_filter, list):
+                for org in org_filter:
+                    org_filter_set.add(org)
+            elif org_filter:
+                org_filter_set.add(org_filter)
+
+        return org_filter_set
+
+    def set_config_by_domain(self, domain):
+        """
+        For a given request domain, find a match in our microsite configuration
+        and then assign it to the thread local in order to make it available
+        to the complete Django request processing
+        """
+        if not self.has_configuration_set() or not domain:
+            return
+        config, tenant_key = self.get_config_by_domain(domain)
+
+        if config and tenant_key:
+            self._set_config_from_obj(domain, config, tenant_key)
+            return
+
+        # If required, delegate to the old microsite backend method
+        return super(TenantConfigCompatibleMicrositeBackend, self).set_config_by_domain(domain)
+
+    def get_all_config(self):
+        """
+        This returns all configuration for all microsites
+        """
+        config = super(TenantConfigCompatibleMicrositeBackend, self).get_all_config()
+
+        # Apply also tenant configs (temporary compat layer)
+        for microsite in self.iterate_sites_with_tenant():
+            values = microsite.lms_configs
+            config[microsite.external_key] = values
```

Comment:
You can merge this and the previous line, since values is only used in this line. @diegomillan